devbox/provision/provision.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+readonly GRAPL_ROOT="${PWD}"
+readonly GRAPL_DEVBOX_DIR="${HOME}/.grapl_devbox"
+mkdir -p "${GRAPL_DEVBOX_DIR}"
+
+########################################
+# Helper functions
+########################################
+has_key() {
+    local -r input_json="${1}"
+    local -r key="${2}"
+    jq --exit-status ".[\"${key}\"]" <<< "${input_json}" > /dev/null
+}
+
+########################################
+# Ensure we're in the Pulumi-enabled venv
+########################################
+if [ ! -f build-support/venv/bin/activate ]; then
+    echo "Set up your virtualenv with 'build-support/manage_virtualenv.sh'"
+    exit 42
+fi
+# shellcheck disable=SC1091
+source build-support/venv/bin/activate
+
+########################################
+# Set up an SSH key specifically for the Devbox
+########################################
+
+readonly SSH_PRIVATE_KEY_FILE="${GRAPL_DEVBOX_DIR}/ssh_key"
```

Comment:
can we name this key something like devbox_ed25519? That way its clear what this is for and what format the ssh key is in